### PR TITLE
Browser deactivation logging

### PR DESCRIPTION
**Pull request by nrajan002c**
Signed-off-by: nganes961 <nambirajan_ganesan@comcast.com>

### CODE REVIEW

File: WebKitBrowser/WebKitBrowser.cpp [hunk 1]
```diff
@@ -350,6 +351,7 @@ namespace Plugin {
     void WebKitBrowser::Deactivated(RPC::IRemoteConnection* connection)
     {
         if (connection->Id() == _connectionId) {
+            TRACE(Trace::Error, (_T("WebKitBrowser::Deactivated: { \"URL\": %s }"), _lastURL.substr(0,80).c_str()));
```
Comment (emutavchi) [inline WebKitBrowser/WebKitBrowser.cpp:354]: I think You can avoid a mem copy with '%.*s' format and specifying limit just before _lastURL.c_str(). i.e.:

```
TRACE(Trace::Error, (_T("WebKitBrowser::Deactivated: { \"URL\": %.*s }"), 80, _lastURL.c_str()));
```

could you please give it a try?

Comment (nrajan002c) [reply WebKitBrowser/WebKitBrowser.cpp]: @emutavchi , the above format is working fine.

**Review by emutavchi — APPROVED** [review]
Looks good to me. @MFransen69  could you please take a look?

File: WebKitBrowser/WebKitBrowser.cpp [hunk 2]
```diff
@@ -309,6 +309,7 @@ namespace Plugin {
     {
         string message(string("{ \"url\": \"") + URL + string("\", \"loaded\": ") + (loaded ? string("true") : string("false")) + string(" }"));
         TRACE(Trace::Information, (_T("URLChanged: %s"), message.c_str()));
+        _lastURL = URL;
```
Comment (MFransen69) [inline WebKitBrowser/WebKitBrowser.cpp:312]: I would only do the string copy (relatively expensive, certainly for bigger strings) when tracing is enabled for this category (the TRACE_ENABLED(CATEGORY) macro can be used to find that out).

And although I do not see any problem with this PR in itself is what is the actual use case if I may ask? I guess tracing the url that caused Webkit to crash? Or would the error tracing category be always enabled so automated action can be taken?

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: WebKitBrowser/WebKitBrowser.cpp (hunk 2 above)
Comment (nrajan002c) [reply WebKitBrowser/WebKitBrowser.cpp:312]: @MFransen69 
The reason for this implementation is, to track the browser crashes by app URL. 
Use case is, We need to know the number of crashes per app basis. 
And we need this log to be printed always.
I can convert (_lastURL =URL) to strcpy.

Comment (MFransen69) [reply WebKitBrowser/WebKitBrowser.cpp:312]: Ah, okay in that case I think you do not want to use Tracing but SYSLOG, correct? Trace lines will only be outputted when for that module and category they are turned on, therefore I suggested to only do the string assignment (which is the same as a strcpy, std::string is not refcounted) when tracing is actually turned on here,  it is off by default. 

But seen your usecase is it perhaps an idea to add this functionality to the PerformanceMetrics plugin, that seems a more logical place then? All the info you need would be available there.
I would be happy to add that functionality there or to assist you in adding it there.

**Comment by anand-ky** [comment]
Closing older PRs. Pls bring changes to latest sprint.

**Pull request closed without merging**